scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,189 @@
+                    parent_key_field_name="parent_id",
+                    source_context="/document/pages/*",
+                    mappings=[
+                        InputFieldMappingEntry(name="content", source="/document/pages/*"),
+                        InputFieldMappingEntry(name="embedding", source="/document/pages/*/vector"),
+                        InputFieldMappingEntry(name="sourcepage", source="/document/metadata_storage_name"),
+                    ],
+                ),
+            ],
+            parameters=SearchIndexerIndexProjectionsParameters(
+                projection_mode=IndexProjectionMode.SKIP_INDEXING_PARENT_DOCUMENTS
+            ),
+        )
+
+        skillset = SearchIndexerSkillset(
+            name=skillset_name,
+            description="Skillset to chunk documents and generating embeddings",
+            skills=[split_skill, embedding_skill],
+            index_projections=index_projections,
+        )
+
+        return skillset
+
+    async def setup(self, search_info: SearchInfo):
+        search_manager = SearchManager(
+            search_info,
+            self.search_analyzer_name,
+            self.use_acls,
+            True,
+            self.embeddings,
+            search_images=False,
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        await search_manager.create_index(
+            [
+                AzureOpenAIVectorizer(
+                    name="myOpenAI",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=AzureOpenAIParameters(
+                        resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+                        deployment_id=self.embeddings.open_ai_deployment,
+                        auth_identity=None,
+                    ),
+                ),
+            ]
+        )
+
+        # create indexer client
+        ds_client = search_info.create_search_indexer_client()
+        ds_container = SearchIndexerDataContainer(name=self.blob_manager.container)
+        data_source_connection = SearchIndexerDataSourceConnection(
+            name=f"{search_info.index_name}-blob",
+            type="azureblob",
+            connection_string=self.blob_manager.get_managedidentity_connectionstring(),
+            container=ds_container,
+        )
+
+        await ds_client.create_or_update_data_source_connection(data_source_connection)
+        print("Search indexer data source connection updated.")
+
+        embedding_skillset = await self.createEmbeddingSkill(search_info.index_name)
+        await ds_client.create_or_update_skillset(embedding_skillset)
+
+    async def run(self, search_info: SearchInfo):
+        if self.document_action == DocumentAction.Add:
+            files = self.list_file_strategy.list()
+            async for file in files:
+                try:
+                    await self.blob_manager.upload_blob(file)
```

Comment:
So if they use DataLake, would this end up making a copy of the DataLake file on Blob Storage? Or is this not compatible with DataLake?

Reply:
sample is to show blob that indexer can access and ingest data. Indexer can also connect to datalake. Use needs to update the datasource connection.